src/fldeff_teleport.c

```diff
@@ -0,0 +1,55 @@
+#include "global.h"
+#include "field_effect.h"
+#include "task.h"
+
+extern bool8 is_light_level_1_2_3_or_6(u8 mapType);
+extern void* hm_add_c3_launch_phase_2(void);
+extern void hm_teleport_run_dp02scr(void);
+extern void* gUnknown_0203CEEC;
+extern void* gUnknown_03005DB0;
+extern void sub_808469C(void);
+extern u8 GetCursorSelectionMonId(void);
+extern void sub_817C94C(void);
+extern u8 oei_task_add(void);
+extern void FieldEffectActiveListRemove(u8 id);
+extern void sub_80B7FC8(void);
+extern void SetPlayerAvatarTransitionFlags(u8);
+
+
+// 817C8BC
+bool8 SetUpFieldMove_Teleport(void)
+{
+    if (is_light_level_1_2_3_or_6(gMapHeader.mapType) == TRUE)
+    {
+        gUnknown_03005DB0 = hm_add_c3_launch_phase_2;
```

Comment:
Looking at pokeruby, `is_light_level_1_2_3_or_6` should be renamed to `Overworld_MapTypeAllowsTeleportAndFly`, `hm_add_c3_launch_phase_2` should be renamed to `FieldCallback_Teleport`, and `gUnknown_03005DB0` should be renamed to `gFieldCallback`.